state/state.go

```diff
@@ -188,10 +188,17 @@ func StateFromProto(pb *tmstate.State) (*State, error) { //nolint:golint
 	state.ChainID = pb.ChainID
 	state.InitialHeight = pb.InitialHeight
 
-	bi, err := types.BlockIDFromProto(&pb.LastBlockID)
-	if err != nil {
-		return nil, err
+	var bi *types.BlockID
+	if pb.LastBlockID.DataAvailabilityHeader != nil {
```

Comment:
Moving this check down to BlockIDFromProto would make the code cleaner

Reply:
ahh yes, but then BlockIDFromProto would do more than just convert to and from proto, and that felt like a leaky abstraction waiting to happen so decided against that.

Reply:
Ok